oauth2client/contrib/gce.py

```diff
@@ -48,28 +45,67 @@
 """
 
 
-def _get_service_account_email(http_request=None):
-    """Get the GCE service account email from the current environment.
-
+def _get_metadata(http_request=None, path=None, recursive=True):
```

Comment:
There are quite a lot of calls to `_get_metadata` below, it seems this will slow down apps unnecessarily. Why not cache results or do more work in the constructor?